### PR TITLE
Merge paradox/scala/io-tcp.md and java/io-tcp.md 

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #23052

## Rendered HTML:

- https://richard-akka-doc.firebaseapp.com/scala/io-tcp.html
- https://richard-akka-doc.firebaseapp.com/java/io-tcp.html

## I split this in two commits:

- **1st commit:** Identical contents in scala/io-tcp.md and java/io-tcp.md -> **this is to take the diff both in scala/io-tcp.md and java/io-tcp.md**

  - [scala/io-tcp.md  (prev) vs. scala/io-tcp.md (after)](https://github.com/richard-imaoka/akka/commit/de6e2007a40e0db1935dce7ac8b38fc268567c08#diff-5c2f49192b560945210a9e5b9a079439)
  - [java/io-tcp.md (prev) vs. java/io-tcp.md (after)](https://github.com/richard-imaoka/akka/commit/de6e2007a40e0db1935dce7ac8b38fc268567c08#diff-1d02342cce8fecfa70080c59edd0d098)

- **2nd commit:** Replace java/io-tcp.md with a symlink to scala/io-tcp.md

### CODE REVIEW

**Comment by patriknw** [comment]
PLS BUILD

File: akka-docs/src/main/paradox/java/io-tcp.md [hunk 1]
```diff
@@ -1,326 +0,0 @@
-# Using TCP
-
-The code snippets through-out this section assume the following imports:
-
-@@snip [IODocTest.java]($code$/java/jdocs/io/japi/IODocTest.java) { #imports }
-
-All of the Akka I/O APIs are accessed through manager objects. When using an I/O API, the first step is to acquire a
-reference to the appropriate manager. The code below shows how to acquire a reference to the `Tcp` manager.
-
-@@snip [EchoManager.java]($code$/java/jdocs/io/japi/EchoManager.java) { #manager }
```
Comment (patriknw) [inline akka-docs/src/main/paradox/java/io-tcp.md]: what about those?

Comment (richard-imaoka) [inline akka-docs/src/main/paradox/java/io-tcp.md]: Is the question where this is in the combined .md file?

```
-@@snip [EchoManager.java]($code$/java/jdocs/io/japi/EchoManager.java) { #manager }
```

If so, it is found in the [diff from the 1st git commit](https://github.com/richard-imaoka/akka/commit/de6e2007a40e0db1935dce7ac8b38fc268567c08#diff-5c2f49192b560945210a9e5b9a079439).

However, 

image

somehow "Files Changed" of the entire pull request doesn't show the diff from scala/io-tcp.md. 

This has happened sometimes when changing java/***.md to a symlink pointing to scala/***.md. Even if I squash two commits into one, this still happens. I don't know why...

Comment (raboof) [inline akka-docs/src/main/paradox/java/io-tcp.md]: Yes github appears to be confused showing diffs that contain symlinks sometimes... viewing the individual commits sometimes helps.

**Review by raboof — APPROVED** [review]

**Pull request merged**